engine/function/itemparser/itemlistparser.php

```diff
@@ -1,14 +1,35 @@
 <?php 
+
 /* Returns a PHP array $id => 'name'
 	 $items = getItemList();
 	 echo $items[2160]; // Returns 'Crystal Coin'
-	*/
-function getItemList() {
-	$item_list = explode(PHP_EOL, file_get_contents('item_list.txt'));
-	$ia = array();
-	for ($i = 0; $i < count($item_list) - 1; $i++) {
-		$it = explode('@', $item_list[$i]);
-		$ia[(int)$it[0]] = ucfirst($it[1]);
+*/
+
+function getItemList() {	
+	$items = parseItems();	
+	return $items;
+}
+
+function getItemById($id) {
+	$items = parseItems();
+	if(isset($items[$id])) {
+		return $items[$id];
 	}
-	return $ia;
-}
\ No newline at end of file
+	return false;
+}
+
+function parseItems() {
+	global $config;
+
+	$items = simplexml_load_file($config['server_path'].'/data/items/items.xml');
+	$out = array();
+	
+	// Create our parsed item list
+	foreach ($items->children() as $item) {
+		if ($item['id'] && $item['name'] != NULL) {
+			$out[(string)$item['id']] = (string)$item['name'];
```

Comment:
Why do we need ID to be a string? ID is represented by an integer afaik?
 Will `$item['name']` ever not be a string? Once again, why do we need to retype?

Reply:
You are right about the id field my bad. And the string conversion is just to stop warnings from using the simple xml object directly as an array item

Reply:
Gotcha, thanks for clarifying.